packages/compass-query-bar/src/components/query-option/query-option.tsx

```diff
@@ -0,0 +1,156 @@
+);
+
+const numericTextInputStyles = css({
+  input: {
+    borderColor: 'transparent',
+  },
+});
+
+const optionInputWithErrorStyles = css({
+  input: {
+    borderColor: uiColors.red.base,
+  },
+});
+
+const queryOptionLabelContainerStyles = css({
+  whiteSpace: 'nowrap',
+  textTransform: 'capitalize',
+  alignItems: 'center',
+  display: 'flex',
+  margin: 0,
+  marginRight: spacing[2],
+});
+
+type QueryOptionProps = {
+  autoPopulated: boolean;
+  hasError: boolean;
+  onChange: (value: string) => void;
+  onApply: () => void;
+  placeholder?: string;
+  queryOption: QueryOptionType;
+  refreshEditorAction: Listenable;
+  schemaFields: string[];
+  serverVersion: string;
+  value?: string | number;
+};
+
+const QueryOption: React.FunctionComponent<QueryOptionProps> = ({
+  autoPopulated,
+  hasError,
+  onApply,
+  onChange,
+  placeholder = '',
+  queryOption,
+  refreshEditorAction,
+  schemaFields = [],
+  serverVersion,
+  value = '',
+}) => {
+  const isAutoCompleteInput = useMemo(
+    () => OPTION_DEFINITION[queryOption].type === 'document',
+    [queryOption]
+  );
+
+  return (
+    <div
+      className={cx(
+        queryOptionStyles,
+        isAutoCompleteInput ? autocompleteOptionStyles : numericOptionStyles
+      )}
+      data-test-id="query-bar-option"
+    >
+      {queryOption !== 'filter' && (
+        <div
+          className={queryOptionLabelContainerStyles}
+          data-test-id="query-bar-option-label"
+        >
+          <Label
+            htmlFor={`query-bar-option-input-${queryOption}`}
+            id={`query-bar-option-input-${queryOption}-label`}
+            className={queryOptionLabelStyles}
+          >
+            {queryOption}
+          </Label>
+        </div>
+      )}
+      <div className={cx(isAutoCompleteInput && autocompleteOptionInputStyles)}>
+        {isAutoCompleteInput ? (
+          <OptionEditor
+            autoPopulated={autoPopulated}
+            hasError={hasError}
+            id={`query-bar-option-input-${queryOption}`}
+            queryOption={queryOption}
+            onApply={onApply}
+            onChange={onChange}
+            placeholder={placeholder}
+            refreshEditorAction={refreshEditorAction}
+            schemaFields={schemaFields}
+            serverVersion={serverVersion}
+            value={value}
+          />
+        ) : (
+          <TextInput
+            aria-labelledby={`query-bar-option-input-${queryOption}-label`}
+            id={`query-bar-option-input-${queryOption}`}
+            data-test-id="query-bar-option-input"
+            className={cx(
+              numericTextInputStyles,
+              hasError && optionInputWithErrorStyles
+            )}
+            type="text"
+            state={hasError ? 'error' : 'none'}
+            value={`${value}`}
+            onChange={(evt: React.ChangeEvent<HTMLInputElement>) =>
+              onChange(evt.target.value)
+            }
+            placeholder={placeholder}
+          />
```

Comment:
Are these so big by design?

![image](https://user-images.githubusercontent.com/5036933/172557524-c34e2894-9601-446a-a0a8-b16c7b35e80d.png)

Reply:
Pinged Claudia on this, currently they're the default size variant from leafygreen.
I'm thinking `small` looks a bit better (before/after):
![Screen Shot 2022-06-08 at 5 40 33 PM](https://user-images.githubusercontent.com/1791149/172723339-5a3d2a91-cad3-432f-9ba4-3559ec8bc205.png)
![Screen Shot 2022-06-08 at 5 47 40 PM](https://user-images.githubusercontent.com/1791149/172723340-9e93ab9d-f201-4f5b-a236-576115551a69.png)
Pushed up the smaller one for now, might change based on feedback.

Reply:
Claudia likes the smaller too! Let's keep it - do you like it?

Reply:
Yeah, does look better to me too 👍